apps/andi/test/integration/andi_web/live/ingestion_live_view/extract_steps/extract_auth_step_form_test.exs

```diff
@@ -3,221 +3,192 @@ defmodule AndiWeb.ExtractAuthStepFormTest do
+      form_data = %{
+        "destination" => new_destination
+      }
 
-      assert {:ok, view, html} = live(conn, @url_path <> ingestion.id)
+      view
+      |> form("##{auth_step.id}", form_data: form_data)
+      |> render_change()
 
-      assert html |> find_elements(".url-form__source-headers-key-input") |> Enum.empty?()
-      assert html |> find_elements(".url-form__source-headers-value-input") |> Enum.empty?()
+      html = render(view)
+      assert get_value(html, "##{auth_step.id}_auth_destination") == new_destination
     end
-  end
-
-  test "required url field displays proper error message", %{conn: conn} do
-    context = %{"url" => "123.com"}
-    {:ok, ingestion} = IngestionHelpers.create_with_auth_extract_step(context) |> IngestionHelpers.save_ingestion()
-
-    extract_step_id = IngestionHelpers.get_extract_step_id(ingestion, 0)
-
-    assert {:ok, view, html} = live(conn, @url_path <> ingestion.id)
-    extract_steps_form_view = find_live_child(view, "extract_step_form_editor")
-    es_form = element(extract_steps_form_view, "#step-#{extract_step_id} form")
 
-    form_data = %{"url" => ""}
-
-    html = render_change(es_form, %{"form_data" => form_data})
-
-    assert get_text(html, "#url-error-msg") == "Please enter a valid url - including http:// or https://"
-  end
-
-  data_test "invalid #{field} displays proper error message", %{conn: conn} do
-    context = %{
-      url: "123.com",
-      body: "",
-      headers: %{"api-key" => "to-my-heart"},
-      cacheTtl: 1_000,
-      destination: "dest",
-      path: ["blah", "blah"]
-    }
-
-    {:ok, ingestion} = IngestionHelpers.create_with_auth_extract_step(context) |> IngestionHelpers.save_ingestion()
-
-    extract_step_id = IngestionHelpers.get_extract_step_id(ingestion, 0)
-
-    assert {:ok, view, html} = live(conn, @url_path <> ingestion.id)
-    extract_steps_form_view = find_live_child(view, "extract_step_form_editor")
-    es_form = element(extract_steps_form_view, "#step-#{extract_step_id} form")
+    test "destination field shows an error if blank", %{
+      view: view,
+      html: html,
+      auth_step: auth_step
+    } do
+      new_destination = ""
 
-    form_data = %{field => value}
+      form_data = %{
+        "destination" => new_destination
+      }
 
-    html = render_change(es_form, %{"form_data" => form_data})
+      view
+      |> form("##{auth_step.id}", form_data: form_data)
+      |> render_change()
 
-    assert get_text(html, "##{field}-error-msg") == error
+      html = render(view)
+      assert get_text(html, "#destination-error-msg") == "Please enter a valid destination."
+    end
 
-    where([
-      [:field, :value, :error],
-      ["headers", %{"0" => %{"key" => "", "value" => "where is it?!"}}, "Please enter valid key(s)."],
-      ["body", "this is invalid json", "Please enter valid JSON"]
-    ])
-  end
+    test "url field can be altered and saved", %{
```

Comment:
Excellent tests. Are these new or moved from somewhere else? Looks like we were missing quite a bit of coverage. Good stuff!